src/Backends/InterfaceSettings.vala

```diff
@@ -57,5 +78,44 @@ public class SettingsDaemon.Backends.InterfaceSettings : GLib.Object {
         accounts_service.cursor_size = interface_settings.get_int (CURSOR_SIZE);
         accounts_service.locate_pointer = interface_settings.get_boolean (LOCATE_POINTER);
         accounts_service.text_scaling_factor = interface_settings.get_double (TEXT_SCALING_FACTOR);
+
+        accounts_service.picture_options = background_settings.get_enum (PICTURE_OPTIONS);
+        accounts_service.primary_color = background_settings.get_string (PRIMARY_COLOR);
+    }
+
+    private void sync_background_to_greeter () {
+        // File.new_for_uri creates file with broken get_basename method, so do this
+        var wallpaper_uri = background_settings.get_string (PICTURE_URI);
+        var wallpaper_path = File.new_for_uri (wallpaper_uri).get_path ();
+        var source = File.new_for_path (wallpaper_path);
+
+        var greeter_data_dir = Path.build_filename (Environment.get_variable ("XDG_GREETER_DATA_DIR"), "wallpaper");
+        if (greeter_data_dir == "") {
+            greeter_data_dir = Path.build_filename ("/var/lib/lightdm-data/", Environment.get_user_name (), "wallpaper");
+        }
+        var folder = File.new_for_path (greeter_data_dir);
+        var dest = File.new_for_path (Path.build_filename (greeter_data_dir, source.get_basename ()));
```

Comment:
you can simplify this by making greeter_data_dir fallback to "/var/lib/lightdm-data/$user", and then creating the wallpaper folder with `File.new_build_filename()`.

dest isn't needed, since we can use `folder.get_child()` when it's actually used.

```suggestion
        var greeter_data_dir = Environment.get_variable ("XDG_GREETER_DATA_DIR") ?? Path.build_filename ("/var/lib/lightdm-data", Environment.get_user_name ());
        var folder = File.new_build_filename (greeter_data_dir, "wallpaper");
```